packages/core/src/node/messaging/messaging-service.ts

```diff
@@ -54,3 +54,18 @@ export namespace MessagingService {
         configure(service: MessagingService): void;
     }
 }
+
+export interface WebSocketConnection extends IConnection {
+    channel: WebSocketChannel;
+}
+export namespace WebSocketConnection {
+    export function is(connection: IConnection): connection is WebSocketConnection {
+        return (connection as WebSocketConnection).channel instanceof WebSocketChannel;
+    }
+
+    export function create(connection: IConnection, channel: WebSocketChannel): WebSocketConnection {
+        const result = connection as WebSocketConnection;
+        result.channel = channel;
```

Comment:
Conceptually speaking, is this not slightly bogus? Technically, there is supposed to be `n` channels for 1 given connection, so should you rather convert this `channel` property into a `channels` list, pushing new channels into it, and closing them all on close?

Reply:
No, every `IConnection` that is established through a WebSocketChannel has exactly one associated channel. `IConnection` is not the raw WebSocket connection, but rather a wrapper around a reader and a writer for a single channel.

We should eliminate this helper method by moving the new `WebSocketConnection` interface to `vscode-ws-jsonrpc`, but IMO that could be done after merging this.

Reply:
Ok, got confused by the name. Can you rename it `WebSocketChannelConnection`?